pallets/resource-commenting/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+// pub use crate::weights::WeightInfo;
+//
+#[cfg(test)]
+mod mock;
+//
+// #[cfg(test)]
+// mod tests;
+//
+// #[cfg(feature = "runtime-benchmarks")]
+// mod benchmarking;
+//
+// pub mod weights;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::{dispatch::RawOrigin, pallet_prelude::*, traits::Currency, transactional};
+    use frame_system::pallet_prelude::*;
+    use sp_runtime::traits::{StaticLookup, Zero};
+    use sp_std::vec::Vec;
+
+    use pallet_posts::{NextPostId, PostExtension};
+    use subsocial_support::{Content, PostId, SpaceId};
+    use sp_std::{convert::TryInto, fmt::Debug};
+
+    // use crate::weights::WeightInfo;
+
+    type ResourceId<T> = BoundedVec<u8, <T as Config>::MaxResourcesIdLength>;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_posts::Config + pallet_spaces::Config {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// The id of the space, where all resource posts will reside.
+        type ResourcesSpaceId: Get<SpaceId>;
+
+        /// The maximum number of characters in a resource id.
+        #[pallet::constant]
+        type MaxResourcesIdLength: Get<u32>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        ResourcePostCreated { resource_id: ResourceId<T>, post_id: PostId },
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        ResourcePostAlreadyCreated,
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn resource_post)]
+    pub type ResourcePost<T: Config> = StorageMap<_, Twox64Concat, ResourceId<T>, PostId>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(Weight::from_ref_time(10_000))]
+        #[transactional]
+        pub fn create_resource_post(
```

Comment:
There should be 2 extrinsics:
- link_post_to_resource (origin, resource_id, post_id)
- create_resource_discussion (origin, resource_id, space_id, Option<first_comment_content>)

Reply:
link_post_to_resource (origin, resource_id, post_id) - just link resource to post (just check that account own the post)

create_resource_discussion (origin, resource_id, space_id) - create new post and link resource (check that account own the space)

Reply:
resource_id, account_id (of the poster)